machinist/node.go

```diff
@@ -0,0 +1,161 @@
+package machinist
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"github.com/enfabrica/enkit/lib/client/ccontext"
+	"github.com/enfabrica/enkit/machinist/rpc/machinist"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"time"
+)
+
+type NodeErr struct {
+}
+type Node struct {
+	Client      machinist.ControllerClient
+	TLSCert     string
+	Connection  *grpc.ClientConn
+	KillChannel chan NodeErr
+	PongChannel chan *machinist.PollResponse
+	*ccontext.Context
+}
+
+func NewNode(nm ...NodeModifier) (*Node, error) {
+	n := &Node{
+		KillChannel: make(chan NodeErr),
+		Context:     ccontext.DefaultContext(),
+	}
+	for _, mod := range nm {
+		if err := mod(n); err != nil {
+			fmt.Println("there was an error", err.Error())
+			return nil, err
+		}
+	}
+	return n, nil
+}
+
+type NodeModifier func(node *Node) error
+
+func (n *Node) listenForPong() {
+	ctx := context.Background()
+	pollStream, err := n.Client.Poll(ctx)
+	if err != nil {
+		n.Kill(err)
+	}
+	for {
+		pollResponse, err := pollStream.Recv()
+		if err != nil {
+			n.Kill(err)
+		}
+		n.PongChannel <- pollResponse
+	}
+
+}
+
+func (n *Node) BeginPolling() {
```

Comment:
some of this code was already checked in / semi-working. I was expecting to see a diff over the existing files, rather than entirely new files? Not sure how the new code bodes with the existing one otherwise.